### PR TITLE
Add support for updating `stop` and `descriptor` documents

### PR DESCRIPTION
**Pull request by Kezzsim**
Self descriptive, adds upport for updating all run record types via the suitcase mongo api.

### CODE REVIEW

File: suitcase/mongo_normalized/__init__.py [hunk 1]
```diff
@@ -48,22 +53,28 @@ def __init__(self, metadatastore_db, asset_registry_db,
...
-                             'datum': self._datum_collection}
+        self._run_start_collection = mds_db.get_collection("run_start")
+        self._run_start_collection_revisions = mds_db.get_collection(
+            "run_start_revisions"
+        )
+        self._run_stop_collection = mds_db.get_collection("run_stop")
+        self._run_stop_collection_revisions = mds_db.get_collection(
+            "run_stop_revisions"
+        )
+        self._event_descriptor_collection = mds_db.get_collection("event_descriptor")
```
Comment (danielballan) [inline suitcase/mongo_normalized/__init__.py:64]: We need `event_descriptor_revisions` too.

File: suitcase/mongo_normalized/__init__.py [hunk 2]
```diff
@@ -159,39 +179,46 @@ def update(self, name, doc):
...
         doc: dict
             The new version of the document. Its uid will be used to match it
             to the current version, the one to be updated.
         """
-        if name == 'start':
+        if name in ["start", "stop", "descriptor"]:
+            # Keys and collection names differ slightly between start, stop and descriptor
+            key = "uid" if name == "start" else "run_start"
+            name = f"_event_{name}" if name == "descriptor" else f"_run_{name}"
             event_model.schema_validators[event_model.DocumentNames.start].validate(doc)
```
Comment (danielballan) [inline suitcase/mongo_normalized/__init__.py:192]:
```suggestion
            event_model.schema_validators[getattr(event_model.DocumentNames, name)].validate(doc)
```

File: suitcase/mongo_normalized/__init__.py (hunk 1 above)
Comment (Kezzsim) [reply suitcase/mongo_normalized/__init__.py:67]: Created in b86982141e4d50c272f1370baf6b132b7e269119

File: suitcase/mongo_normalized/__init__.py [hunk 3]
```diff
@@ -159,39 +182,56 @@ def update(self, name, doc):
...
+            old = current_col.find_one({key: doc[key]})
+            if old is None and (name == "_run_stop" or name == "_event_descriptor"):
+                # New stop or descriptor document : insert it
+                current_col.insert_one(doc)
+            else:
+                old.pop("_id")
+                # Field Saftey Enforcement : Prevent restricted fields from changing
+                restricted_fields = ["run_start"]
+                for field in restricted_fields:
+                    if field in old and field in doc:
```
Comment (danielballan) [inline suitcase/mongo_normalized/__init__.py:209]: I think this would permit _removing_ the `run_start` key.

**Pull request merged**